.gitlab/includes/clang11_pipeline.yml

```diff
@@ -19,77 +19,28 @@ include:
     CMAKE_FLAGS: "-DCMAKE_BUILD_TYPE=$BUILD_TYPE -DPIKA_WITH_CXX_STANDARD=$CXXSTD \
                   -DPIKA_WITH_MAX_CPU_COUNT=256 -DPIKA_WITH_MALLOC=system \
                   -DPIKA_WITH_SPINLOCK_DEADLOCK_DETECTION=ON"
+    CSCS_REBUILD_POLICY: always
 
 clang11_spack_compiler_image:
-  stage: spack_compiler
-  needs: [base_spack_image]
   extends:
-    - .container-builder
+    - .compiler_image_template
     - .variables_clang11_config
```

Comment:
Just checking for my own understanding: Are these sensitive to the order in which blocks are listed under `extends:`? If later blocks override/use earlier blocks, would it make sense (just for "sanity") to put the `variables_clang11_config` block (and others) always first since they don't depend on other variables? I don't know if that's always the case thought.

The same comment applies of course to other configurations and the other stages as well.

Reply:
What happens if a variable is redefined (we might e.g. want defaults for some things)? Or if a variable depends on another variable?

Reply:
Thanks for checking and the link! I was thinking of this case specifically: https://docs.gitlab.com/ee/ci/yaml/yaml_optimization.html#merge-details, but all of this is good to have here for reference.